code/datums/brain_damage/mrat.dm

```diff
@@ -0,0 +1,156 @@
+		icons_available += list("[T.name]" = image(icon = T.icon, icon_state = T.icon_state))
+
+/mob/camera/imaginary_friend/mrat/proc/Costume()
+	update_available_icons()
+	if(icons_available)
+		var/selection = show_radial_menu(src, src, icons_available, radius = 38)
+		if(!selection)
+			return
+
+		for(var/datum/mrat_type/T in mrat_types)
+			if(T.name == selection)
+				current_costume = T
+				human_image = image(icon = T.icon, icon_state = T.icon_state)
+				color = T.color
+				Show()
+				return
+
+/mob/camera/imaginary_friend/mrat/proc/PickName()
+	var/picked_name = sanitize_name(stripped_input(src, "Enter your mentor rat's name", "Rat Name", "Mentor Rat", MAX_NAME_LEN - 3 - length(key)))
+	if(!picked_name || picked_name == "")
+		picked_name = "Mentor Rat"
+	log_game("[key_name(src)] has set \"[picked_name]\" as their mentor rat's name for [key_name(owner)]")
+	name = "[picked_name] ([key])"
+
+/mob/camera/imaginary_friend/mrat/friend_talk()
+	. = ..()
+	if(!current_costume || !istype(current_costume))
+		return
+	SEND_SOUND(owner, sound(current_costume.sound, volume=current_costume.volume))
+	SEND_SOUND(src, sound(current_costume.sound, volume=current_costume.volume))
+
+/mob/camera/imaginary_friend/mrat/greet()
+	to_chat(src, "<span class='notice'><b>You are the mentor rat of [owner]!</b></span>")
+	to_chat(src, "<span class='notice'>Do not give [owner] any OOC information from your time as a ghost.</span>")
+	to_chat(src, "<span class='notice'>Your job is to answer [owner]'s question(s) and you are given this form to assist in that.</span>")
+	to_chat(src, "<span class='notice'>Don't be stupid with this or you will face the consequences.</span>")
+
+/mob/camera/imaginary_friend/mrat/Initialize(mapload, _trauma)
+	. = ..()
+	costume = new
+	costume.Grant(src)
+	leave = new
+	leave.Grant(src)
+	grant_all_languages()
+
+/mob/camera/imaginary_friend/mrat/setup_friend()
+	human_image = null
+
+/datum/action/innate/mrat_costume
+	name = "Change Appearance"
+	desc = "Shape your appearance to whatever you desire."
+	icon_icon = 'icons/mob/actions/actions_minor_antag.dmi'
+	background_icon_state = "bg_revenant"
+	button_icon_state = "ninja_phase"
+
+/datum/action/innate/mrat_costume/Activate()
+	var/mob/camera/imaginary_friend/mrat/I = owner
+	I.Costume()
+
+/datum/action/innate/mrat_leave
+	name = "Leave"
+	desc = "Leave and return to your ghost form."
+	icon_icon = 'icons/mob/actions/actions_minor_antag.dmi'
+	background_icon_state = "bg_revenant"
+	button_icon_state = "beam_up"
+
+/datum/action/innate/mrat_leave/Activate()
+	var/mob/camera/imaginary_friend/I = owner
+	if(alert(I, "Are you sure you want to leave?", "Leave:", "Yes", "No") != "Yes")
+		return
+	to_chat(I, "<span class='warning'>You have ejected yourself from [I.owner].</span>")
+	to_chat(I.owner, "<span class='warning'>Your mentor has left.</span>")
+	qdel(I.trauma)
+
+/mob/camera/imaginary_friend/mrat/pointed(atom/A as mob|obj|turf in view())
```

Comment:
What's this for?

Reply:
I believe this removes the parent behavior of copying things you point at.

Reply:
Nevermind, it's so the pointing is only visible to the person you are mentoring.
`/mob/camera` doesn't inherit this proc from `/mob/living`
so it's emulating it

```
/mob/living/pointed(atom/A as mob|obj|turf in view())
	if(incapacitated())
		return FALSE
	if(HAS_TRAIT(src, TRAIT_DEATHCOMA))
		return FALSE
	if(!..())
		return FALSE
	visible_message("<b>[src]</b> points at [A].", "<span class='notice'>You point at [A].</span>")
	return TRUE
```